x-pack/legacy/plugins/ml/public/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -481,17 +488,22 @@ export class TimeSeriesExplorer extends React.Component {
 
     this.contextChartSelectedInitCallDone = false;
 
+    // Only when `fullRefresh` is true we'll reset all data
+    // and show the loading spinner within the page.
     this.setState({
-      chartDetails: undefined,
-      contextChartData: undefined,
-      contextForecastData: undefined,
-      focusChartData: undefined,
-      focusForecastData: undefined,
+      fullRefresh,
       loadCounter: currentLoadCounter + 1,
       loading: true,
-      modelPlotEnabled: isModelPlotEnabled(currentSelectedJob, +currentDetectorId, currentEntities),
-      hasResults: false,
-      dataNotChartable: false
+      ...(fullRefresh ? {
+        chartDetails: undefined,
```

Comment:
Component's state is pretty massive, maybe chart related props deserve a dedicated method?

Reply:
[This PR](https://github.com/elastic/kibana/pull/41739) has a note about the history of this file, it's a recent mostly as-is conversion from angularjs to React.